packages/ra-ui-materialui/src/list/filter/FilterListItem.tsx

```diff
@@ -153,14 +153,20 @@ const FilterListItem: FC<{ label: string; value: any }> = props => {
     );
 
     const addFilter = () => {
-        setFilters({ ...filterValues, ...value }, null);
+        setFilters({ ...filterValues, ...value }, null, false);
     };
+
     const removeFilter = () => {
-        const inverseValue = Object.keys(value).reduce((acc, key) => {
-            acc[key] = undefined;
-            return acc;
-        }, {} as any);
-        setFilters({ ...filterValues, ...inverseValue }, null);
+        const keysToRemove = Object.keys(value);
+        const filters = Object.keys(filterValues).reduce(
+            (acc, key) =>
+                keysToRemove.includes(key)
+                    ? acc
+                    : { ...acc, [key]: filterValues[key] },
+            {}
+        );
```

Comment:
If you are agree with the previous comment, this one can also be written with a loop:

```suggestion
		const filters = {}
		for(filter in filters) {
			if(keysToRemove.includes(filter)) filters[key] = filterValues[filter]
		}
```